src/main.rs

```diff
@@ -445,6 +464,130 @@ fn analog_write(spi_drv: &mut SpiDrv, uart: &mut EnabledUart, pin: u8, value: u8
     spi_drv.esp_deselect();
 }
 
+fn wifi_set_passphrase(spi_drv: &mut SpiDrv, uart: &mut EnabledUart, mut ssid: String<STR_LEN>, mut passphrase: String<STR_LEN>) -> bool {
+    spi_drv.wait_for_esp_select();
+
+    spi_drv.send_cmd(uart, SET_PASSPHRASE, 2).ok().unwrap();
+
+    // FIXME: for the real crate, don't use unsafe
+    let ssid_bytes: &mut [u8] = unsafe { ssid.as_bytes_mut() };
+    writeln!(uart, "ssid: {:?}\r", ssid_bytes).ok().unwrap();
+    spi_drv.send_param(uart, ssid_bytes, false).ok().unwrap();
+
+    // FIXME: for the real crate, don't use unsafe
+    let passphrase_bytes: &mut [u8] = unsafe { passphrase.as_bytes_mut() };
+    writeln!(uart, "passphrase: {:?}\r", passphrase_bytes).ok().unwrap();
+    spi_drv.send_param(uart, passphrase_bytes, true).ok().unwrap();
+
+    let command_size: u8 = 6 + ssid.len() as u8 + passphrase.len() as u8;
+    spi_drv.pad_to_multiple_of_4(uart, command_size);
+ 
+    spi_drv.esp_deselect();
+    spi_drv.wait_for_esp_select();
+
+    // Wait for reply
+    let data: u8 = 0;
+    let wait_response = spi_drv.wait_response_cmd(uart, SET_PASSPHRASE, 1);
+    match wait_response {
+        Ok(params) => {
+            write!(uart, "\twifi_set_passphrase_response: ").ok().unwrap();
+            for byte in params {
+                let c = byte as char;
+                write!(uart, "{:?}", c).ok().unwrap();
+            }
+            writeln!(uart, "\r\n").ok().unwrap();
+        }
+        Err(e) => {
+            writeln!(uart, "\twifi_set_passphrase_response Err: {:?}\r", e).ok().unwrap();
+            spi_drv.esp_deselect();
+            return false;
+        }
+    }
+
+    spi_drv.esp_deselect();
+
+    true
+}
```

Comment:
Just an idea. Not sure on what is conventional for this sort of thing.
```suggestion
    match wait_response {
        Ok(params) => {
            write!(uart, "\twifi_set_passphrase_response: ").ok().unwrap();
            for byte in params {
                let c = byte as char;
                write!(uart, "{:?}", c).ok().unwrap();
            }
            writeln!(uart, "\r\n").ok().unwrap();

            let success: bool = true;
        }
        Err(e) => {
            writeln!(uart, "\twifi_set_passphrase_response Err: {:?}\r", e).ok().unwrap();
           
            let success: bool = false;
        }
    }
    spi_drv.esp_deselect();
    success
}
```

Reply:
Hmm, I'm not fully sure/convinced on this. That success variable seems for me bit redundant (since we are operating in embedded area). But maybe it would make reading the flow of logic more readable ?. Maybe we could talk this over at some occasion

@jhodapp  what do you think ?

Reply:
I actually do like what @calebbourg suggested here, and I'm also pretty sure that rustc would optimize it to being very similar to putting the literal `true` that you have anyway. I think we're also getting to the point where we can implement more verbose/readable code in the embedded space because the compiles have gotten so good, and even because RAM/ROM space have gotten larger because they're cheaper.

If this were a case where we knew we were just about out of memory and/or it caused a significant timing/performance issue, then your original proposal Borys would be spot on. Just some of my thoughts but I don't feel very strongly about either approach.